src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -2028,6 +2028,7 @@ static void addIdentifiers(CSharpSyntaxNode? node, ConcurrentDictionary<Identifi
                             case GotoStatementSyntax { RawKind: (int)SyntaxKind.GotoStatement }:
                             case TypeParameterConstraintClauseSyntax:
                             case AliasQualifiedNameSyntax:
+                            case AttributeListSyntax:
```

Comment:
>case AttributeListSyntax: [](http://example.com/codeflow?start=28&length=25)

I guess I should have added a comment that logic in this lambda is based on ```Binder.IdentifierUsedAsValueFinder.CheckIdentifiersInNode.childrenNeedChecking```. It can be more permissive (i.e. allow us to dive into more nodes), but should not be more restrictive. Since ```childrenNeedChecking``` has the following case:
```
                        case AttributeListSyntax:
                            // References in attributes, if any, are errors
                            // Skip them
                            return false;
```
I am fine with adding a similar case here with the same comment/explanation. Given the sated goal "Predict all identifiers in the body that should go through Binder.BindIdentifier method", putting it into the bucket of "uninteresting" nodes is somewhat misleading because it contains identifiers that go through Binder.BindIdentifier.

At the same time, I would prefer an alternative approach to suppressing the assert. The one that will allow ```assertBindIdentifierTargets``` to be closer to achieving the stated goal (quoted above). 

```assertBindIdentifierTargets``` already disables asserts in presence of errors since there is no guaranty that we bind all the identifiers that are supposed to be bound. The scenarios that we are dealing with are actually error scenarios, we skip binding the attributes, but report a warning instead. That is why ```assertBindIdentifierTargets``` fails to disable the check. I suggest to fix this flaw instead. I'll make a specific suggestion at the relevant location.